src/lightning/pytorch/loops/evaluation_loop.py

```diff
@@ -304,6 +304,11 @@ def teardown(self) -> None:
         if self._data_fetcher is not None:
             self._data_fetcher.teardown()
             self._data_fetcher = None
+
+        if self._data_source is not None:
+            self._data_source.instance = None
+        self._combined_loader = None
```

Comment:
We could follow the pattern above and add a teardown method on the `_DataLoaderSource`

Reply:
Got it. I've updated the PR; Hope this is what you meant.